src/resolve/resolved-manager.c

```diff
@@ -192,13 +198,46 @@ static int manager_process_address(sd_netlink *rtnl, sd_netlink_message *mm, voi
                 break;
         }
 
+        (void) manager_send_dns_configuration_changed(m, l, /* reset= */ true);
+
         return 0;
 
 fail:
         log_warning_errno(r, "Failed to process RTNL address message: %m");
         return 0;
 }
 
+static int manager_process_route(sd_netlink *rtnl, sd_netlink_message *mm, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+        uint16_t type;
+        uint32_t ifindex = 0;
+        int r;
+
+        assert(rtnl);
+        assert(mm);
+
+        r = sd_netlink_message_get_type(mm, &type);
+        if (r < 0) {
+                log_warning_errno(r, "Failed not get message type, ignoring: %m");
+                return 0;
+        }
+
+        if (!IN_SET(type, RTM_NEWROUTE, RTM_DELROUTE)) {
+                log_warning("Unexpected message type %u when processing route, ignoring.", type);
+                return 0;
+        }
+
+        r = sd_netlink_message_read_u32(mm, RTA_OIF, &ifindex);
+        if (r < 0) {
+                log_warning_errno(r, "Failed to get route ifindex, ignoring: %m");
```

Comment:
Some routes do not have RTA_OIF. Please downgrade the log level to debug when `-ENODATA`.

Reply:
I guess we still need to call `manager_send_dns_configuration_changed()` in such cases. I will re-work that.

Reply:
Ah, yes. Indeed.